File: base/sort/python/SelectionSort.py
# 时间复杂度O(n^2),最坏情况O(n^2)，最好情况O(n^2),空间复杂度O(1),稳定
def selection_sort(unsort_nums):
    for i in range(len(unsort_nums)):
        min_index = i
        for j in range(i + 1, len(unsort_nums)):
            if unsort_nums[j] < unsort_nums[min_index]:
                min_index = j
        if min_index != i:
            unsort_nums[i], unsort_nums[min_index] = unsort_nums[min_index], unsort_nums[i]

File: base/sort/java/SelectionSort.java
package sort.java;

//选择排序
public class SelectionSort {

    public static void selectSort(int[] array) {
        for (int i = 0; i < array.length-1; i++) {
            //每次从剩余没有排序的元素中，找出最小的，放到指定位置
            int minIndex = i;
            for (int j = i + 1; j < array.length; j++) {
                if (array[j] < array[minIndex]) {
                    minIndex = j;
                }
            }
            if (minIndex != i) swap(array, i, minIndex);
        }
    }

    private static void swap(int[] array, int i, int j) {
        int tmp = array[i];
        array[i] = array[j];
        array[j] = tmp;
    }
}
